jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py

```diff
@@ -66,8 +66,8 @@ def vue_calculate_moment(self, *args):
         fname_label = self.dataset_selected.replace("[", "_").replace("]", "")
         self.filename = f"moment{n_moment}_{fname_label}.fits"
 
-        self.add_results.add_results_from_plugin(self.moment)
         self._link_moment_data()
+        self.add_results.add_results_from_plugin(self.moment)
```

Comment:
@javerbukh needs to look at this one since he did a lot of the linking here if I remember correctly.

Reply:
he actually gets credit for pointing me here during code review

Reply:
... but that said, its now failing a test for the number of links, so @javerbukh - can you check that test and see if the new number of links is actually correct and the test needs updating, or if this maybe isn't the whole fix after all 🤔

Reply:
I'll add it to my queue.

Reply:
@kecnry You're going to hate this answer, it's an issue with linking 😄 There probably needs to be some code added to `AddResults` to link new data that is added. I'll make a ticket for that.

Reply:
ok, I'll leave this PR in draft then since it breaks tests and we can just add to it when we get to it?